contracts/Tokens/VTokens/VBep20.sol

```diff
@@ -62,7 +80,25 @@ contract VBep20 is VToken, VBep20Interface {
     // @custom:event Emits Transfer event on success
     // @custom:event Emits RedeemFee when fee is charged by the treasury
     function redeemUnderlying(uint redeemAmount) external returns (uint) {
-        return redeemUnderlyingInternal(redeemAmount);
+        return redeemUnderlyingInternal(msg.sender, msg.sender, redeemAmount);
+    }
+
+    /**
+     * @notice Sender redeems underlying assets on behalf of some other address. This function is only available
+     *   for senders, explicitly marked as delegates of the supplier using `comptroller.updateDelegate`
+     * @dev Accrues interest whether or not the operation succeeds, unless reverted
+     * @param redeemer, on behalf of whom to redeem
+     * @param redeemAmount The amount of underlying to receive from redeeming vTokens
+     * @return uint Returns 0 on success, otherwise returns a failure code (see ErrorReporter.sol for details).
+     */
+    // @custom:event Emits Redeem event on success
+    // @custom:event Emits Transfer event on success
+    // @custom:event Emits RedeemFee when fee is charged by the treasury
+    function redeemUnderlyingBehalf(address redeemer, uint redeemAmount) external returns (uint) {
+        require(comptroller.approvedDelegates(redeemer, msg.sender), "not an approved delegate");
+        address payable receiver = msg.sender;
```

Comment:
Is it needed?

Reply:
It is not needed so I have removed it.

https://github.com/VenusProtocol/venus-protocol/pull/442/commits/c1bcf7a79cf6c64073dc85347c44bc1117ac3c5e